lighthouse-core/gather/drivers/cri.js

```diff
@@ -17,105 +17,115 @@
 'use strict';
 
 const Driver = require('./driver.js');
-const chromeRemoteInterface = require('chrome-remote-interface');
+const EventEmitter = require('events').EventEmitter;
+const WebSocket = require('ws');
+const http = require('http');
 const port = process.env.PORT || 9222;
 
-const log = require('../../lib/log.js');
-
 class CriDriver extends Driver {
   constructor() {
     super();
-
-    /**
-     * Chrome remote interface instance.
-     */
-    this._cri = null;
+    this._lastCommandId = 0;
+    /** @type {!Map<number, function(object)}*/
+    this._callbacks = new Map();
+    this._eventEmitter = new EventEmitter();
   }
 
   /**
    * @return {!Promise<undefined>}
    */
   connect() {
-    return new Promise((resolve, reject) => {
-      if (this._cri) {
-        return resolve();
-      }
-
-      // Make a new tab, stopping Chrome from accidentally giving CRI an "Other" tab.
-      // Also disable the lint check because CRI uses "New" for the function name.
-      /* eslint-disable new-cap */
-      chromeRemoteInterface.New((err, tab) => {
-        if (err) {
-          log.warn('CRI driver', 'cannot create new tab, will reuse tab.', err);
-        }
-
-        chromeRemoteInterface({port: port, chooseTab: tab}, chrome => {
-          this._tab = tab;
-          this._cri = chrome;
-          // The CRI instance is also an EventEmitter, so use directly for event dispatch.
-          this._eventEmitter = chrome;
```

Comment:
This is what prevents me from using it as is in the Connection extraction refactoring.